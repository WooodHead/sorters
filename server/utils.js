function prepare(o) {
    if (o && o._id) {
        o._id = o._id.toString()
    }
    return o
}

async function findOne(Collection, ...args) {
    return prepare(await Collection.findOne(...args))
}

async function find(Collection, ...args) {
    return (await Collection.find(...args).toArray()).map(prepare)
}

module.exports = {
    prepare,
    findOne,
    find,
}