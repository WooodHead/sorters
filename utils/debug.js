
export function log(arg) {
    console.log(arg)
    return arg
}
