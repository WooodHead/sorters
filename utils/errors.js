export const errorMessage = (e) => {
    if (e.graphQLErrors && e.graphQLErrors.length > 0) {
        return e.graphQLErrors[0].message
    } else {
        return e.message
    }
}