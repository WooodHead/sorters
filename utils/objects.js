
function toObject(arr) {
    const res = {}
    for (const [key, value] of arr) {
        res[key] = value
    }
    return res
}

function objectMap(object, keyMap = key => key, valueMap = value => value) {
    return Object.keys(object).reduce((o, key) => {
        o[keyMap(key)] = valueMap(object[key])
        return o
    }, {})
}

module.exports = {
    toObject,
    objectMap
}
