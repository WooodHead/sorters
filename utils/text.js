export const replaceAll = (string, search, replacement) => {
  if (typeof string === "string") {
    return string.replace(new RegExp(search, 'g'), replacement)
  } else {
    return string
  }
}

export const teaserAndInfo = (content, min, opt, max) => {
	if (content) {
		if (content.length <= max) {
			return {
				cut: false,
				teaser: content
			};
		}
		const words = content.split(" ");
		let teaser = "";
		for (let i=0; i<words.length; i++) {
			const candidate = words[i];
			let wouldbelength = teaser.length + 1 + candidate.length + 3;
			if (wouldbelength >= opt) {
				// We're done
				if (wouldbelength > max) {
					if (teaser.length + 3 < min) {
						// cut the word
						return {
							teaser: teaser + " " + candidate.substring(0, opt - teaser.length - 1) + "...",
							cut: true,
						};
					} else {
						return {
							teaser: teaser + "...",
							cut: true,
						};
					}
				} else {
					return {
						teaser: teaser + " " + candidate + "...",
						cut: true,
					};
				}
			}
			teaser += " " + candidate;
		}
		return {
			teaser: teaser + "...",
			cut: true,
		};
	} else {
		return {
			teaser: "",
			cut: false,
		};
	}
};
export const teaser = (content, min, opt, max) => {
	const teaser = teaserAndInfo(content, min, opt, max);
	return teaser.teaser;
}
