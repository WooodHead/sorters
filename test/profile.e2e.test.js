import cheerio from 'cheerio'
import {pretty} from './utils'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import {generateAndLogUser, setUserData} from './fixtures'

let db
let browser

beforeAll(async () => {
    const res = await setup()
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()
})

afterAll(async () => {
    await teardown({db, browser})
})

describe('profile', () => {
    it('displays', async () => {
        const browserPage = await browser.newPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const response = await browserPage.goto(`http://localhost:3000/account/profile`)
        expect(response.status()).toBe(200)

        const text = await response.text()
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })
})
