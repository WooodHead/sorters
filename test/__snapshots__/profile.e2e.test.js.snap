// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile displays 1`] = `
"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"324878792\\">
  <div data-reactid=\\"2\\">
    <!-- react-empty: 3 -->
    <div style=\\"padding-top:50px;padding-bottom:20px;\\" data-reactid=\\"4\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\" data-reactid=\\"5\\">
        <div class=\\"container\\" data-reactid=\\"6\\">
          <div class=\\"navbar-header\\" data-reactid=\\"7\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\" data-reactid=\\"8\\"><span class=\\"sr-only\\" data-reactid=\\"9\\">Toggle navigation</span><span class=\\"icon-bar\\" data-reactid=\\"10\\"></span><span class=\\"icon-bar\\" data-reactid=\\"11\\"></span><span class=\\"icon-bar\\" data-reactid=\\"12\\"></span></button>
            <a class=\\"navbar-brand\\"
              href=\\"/\\" data-reactid=\\"13\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\" data-reactid=\\"14\\">
            <ul class=\\"nav navbar-nav navbar-right\\" data-reactid=\\"15\\">
              <li class=\\"false\\" data-reactid=\\"16\\">
                <a href=\\"/\\" data-reactid=\\"17\\">Home</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"18\\">
                <a href=\\"/about\\" data-reactid=\\"19\\">About</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"20\\">
                <a href=\\"/news\\" data-reactid=\\"21\\">News</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"22\\">
                <a href=\\"/users\\" data-reactid=\\"23\\">Sorters</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"24\\">
                <a href=\\"/dashboard\\" data-reactid=\\"25\\">Dashboard</a>
              </li>
              <li class=\\"dropdownfalse\\" data-reactid=\\"26\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\" data-reactid=\\"27\\">
                  <!-- react-text: 28 -->Hello,
                  <!-- /react-text -->
                  <!-- react-text: 29 -->test
                  <!-- /react-text -->
                  <!-- react-text: 30 -->
                  <!-- /react-text --><span class=\\"caret\\" data-reactid=\\"31\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div style=\\"max-width:400px;margin:auto;\\" data-reactid=\\"32\\">
      <h1 data-reactid=\\"33\\">Profile</h1>
      <div data-reactid=\\"34\\">
        <div data-reactid=\\"35\\">
          <p data-reactid=\\"36\\">
            <!-- react-text: 37 -->Your profile is live at
            <!-- /react-text -->
            <a href=\\"/u/test\\" data-reactid=\\"38\\">
              <!-- react-text: 39 -->/u/
              <!-- /react-text -->
              <!-- react-text: 40 -->test
              <!-- /react-text -->
            </a>
            <!-- react-text: 41 -->.
            <!-- /react-text -->
          </p>
          <div data-reactid=\\"42\\">
            <h2 data-reactid=\\"43\\">Edit Profile</h2>
            <form data-reactid=\\"44\\">
              <div class=\\"form-group\\" data-reactid=\\"45\\">
                <label for=\\"name\\" data-reactid=\\"46\\">Name</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"name\\" value=\\"Test User\\" data-reactid=\\"47\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"48\\">
                <label for=\\"about\\" data-reactid=\\"49\\">About you</label>
                <textarea class=\\"form-control\\" rows=\\"2\\" placeholder=\\"Write a short blurb about yourself.\\" data-reactid=\\"50\\">Test user about</textarea>
              </div>
              <div class=\\"form-group\\" data-reactid=\\"51\\">
                <label for=\\"bio\\" data-reactid=\\"52\\">Bio</label>
                <textarea class=\\"form-control\\" rows=\\"4\\" placeholder=\\"Tell us about yourself in more detail.\\" data-reactid=\\"53\\">Test user bio</textarea>
              </div>
              <div class=\\"form-group\\" data-reactid=\\"54\\">
                <label for=\\"website\\" data-reactid=\\"55\\">Website</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"website\\" value=\\"http://website\\" data-reactid=\\"56\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"57\\">
                <label for=\\"blog\\" data-reactid=\\"58\\">Blog</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"blog\\" value=\\"http://blog\\" data-reactid=\\"59\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"60\\">
                <label for=\\"youtube\\" data-reactid=\\"61\\">Youtube</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"youtube\\" value=\\"http://youtube\\" data-reactid=\\"62\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"63\\">
                <label for=\\"twitter\\" data-reactid=\\"64\\">Twitter</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"twitter\\" value=\\"http://twitter\\" data-reactid=\\"65\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"66\\">
                <label for=\\"reddit\\" data-reactid=\\"67\\">Reddit</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"reddit\\" value=\\"http://reddit\\" data-reactid=\\"68\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"69\\">
                <label for=\\"patreon\\" data-reactid=\\"70\\">Patreon</label>
                <input type=\\"text\\" class=\\"form-control\\" id=\\"patreon\\" value=\\"http://patreon\\" data-reactid=\\"71\\">
              </div>
              <div class=\\"form-group\\" data-reactid=\\"72\\">
                <button type=\\"submit\\" class=\\"btn btn-primary btn-block\\" data-reactid=\\"73\\">Save profile</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <footer data-reactid=\\"74\\">
      <div class=\\"container\\" data-reactid=\\"75\\">
        <hr data-reactid=\\"76\\">
        <p data-reactid=\\"77\\">
          <!-- react-text: 78 -->Brought to you with &#x2764; by
          <!-- /react-text -->
          <a href=\\"/about\\" data-reactid=\\"79\\">Nick Redmark</a>
          <!-- react-text: 80 -->. Support and discuss the development of this platform on
          <!-- /react-text -->
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\" data-reactid=\\"81\\">Patreon</a>
          <!-- react-text: 82 -->.
          <!-- /react-text -->
        </p>
      </div>
    </footer>
  </div>
</div>"
`;
